### DISCUSSION TITLE
API Endpoint to keep car awake deliberately

### DISCUSSION THREAD

**Discussion by Javex** [teslamate-org/teslamate#3231] [Ideas]
This may sound like an odd feature request, but I was wondering if you've considered adding an endpoint to deliberately wake the car & keep it awake for a certain time.

Here's my use case that made me think of this: I use "[teslausb](https://github.com/marcone/teslausb)" which archives footage from the car onto a server when I come, thus preserving dashcam footage beyond the size of a USB stick. The problem is, when you come home and get out of the car, it will go to sleep quicker than the raspberry pi (powered by the USB port in the glovebox) can finish archiving. There are several solutions to this, including providing the Pi with a battery and (my current solution) calling the Tesla API from the Pi to keep the car awake.

While this is working, it has one major drawback: My refresh token for my account is now in the car, thus if someone steals the car they can get to the token. In addition, it's simply another tool that has access to my account.

So I thought: Teslamate already has my token and already has the ability to talk to the API, so what if it provided an endpoint that I can call to simply say "please keep the car awake for X minutes" and then it'll do that and if I hit it again, it'll reset and keep it await for another X minutes. A second endpoint could explicitly stop this to speed up going back to sleep.

Is this something that's possible?

Note that the current implementation in teslausb is very simple: It first wakes the car by calling `/wake_up` and then opens a connection to `https://streaming.vn.teslamotors.com/connect` . It runs both of this on a 90 second schedule until it's done and then just stops and lets the car sleep.

**Comment by wooter** [comment]
This functionality already exists in TeslaUSB, but I believe the authentication method is not compatible with the latest version of the Tesla API. A workaround is to use something like Apple’s Shortcuts app and the WatchforTesla app to activate and deactivate Senry mode, so your car gets power on the USB ports and your raspberry pie is powered up to upload the files. Additionally, you can then add Geo fencing because it doesn’t make sense to have the car wake up when it’s not within range of the Wi-Fi network to upload the files.

Publicising an API would be like TeslaFi, which has opened their own TeslaFi API to control your car, without using the Tesla API directly.
